File: 总题库/21.调整数组顺序使奇数位于偶数前面.py
class Solution:
    def exchange(self, nums: List[int]) -> List[int]:
        if len(nums) < 2:
            return nums;
        
        left = 0
        right = len(nums) - 1

        while left < right:
            while left < right and nums[left] & 1:
                left += 1
            while left < right and (nums[right] & 1) == 0:
                right -= 1
            temp = nums[left]
            nums[left] = nums[right]
            nums[right] = temp

        return nums

File: 总题库/21.调整数组顺序使奇数位于偶数前面.cpp
class Solution {
public:
    vector<int> exchange(vector<int>& nums) {
        if (nums.size() < 2) return nums;

        int left = 0, right = nums.size() - 1;
        while (left < right) {
            while (left < right && nums[left] & 1) {
                left++;
            }
            while (left < right && (nums[right] & 1) == 0) {
                right--;
            }
            swap(nums[left], nums[right]);
        }
        return nums;
    }
};